templates/distribution/manifests/dr/kustomization.yaml.tpl

```diff
@@ -41,13 +61,65 @@ patchesStrategicMerge:
   - patches/velero-schedule-manifests.yml
   - patches/velero-schedule-full.yml
 {{- end }}
+{{- if eq .spec.distribution.common.provider.type "none" }}
+{{- if ne .spec.distribution.modules.dr.etcdBackup.type "none" }}
+  - patches/etcd-backup-schedule.yml
+{{- end }}
+{{- end }}
```

Comment:
you can (should) have this patch in the `patches` above too, simplifying a little the logic.

Notice that with kustomize v5 patchesStrategicMerge has been marked as deprecated and you should switch to patches anyway sooner o later.